.github/workflows/_publish-mobile-github-release.yml

```
@@ -288,6 +300,20 @@ jobs:
           echo '```json' >> $GITHUB_STEP_SUMMARY
           echo "$json" >> $GITHUB_STEP_SUMMARY
           echo '```' >> $GITHUB_STEP_SUMMARY
+          echo "changed_to_state=$changed_to_state" >> $GITHUB_OUTPUT
+
+      - name: Disable workflow if published
+        if: steps.create_workflow_state_artifact.outputs.changed_to_state == 'published'
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          if [ "${{ inputs.dry_run }}" = "true" ]; then
+            echo "Dry run mode - skipping gh workflow disable command"
+            gh workflow list publish-github-release.yml
+          else
+            echo "Disabling workflow to prevent further runs"
+            gh workflow disable publish-github-release.yml
+          fi
```

Same as above about using the inputs:

```
   env:
       WORKFLOW_NAME: ${{ inputs.workflow_name}
```       
```suggestion
          if [ "${{ inputs.dry_run }}" = "true" ]; then
            echo "Dry run mode - skipping gh workflow disable command"
            gh workflow list "$WORKFLOW_NAME"
          else
            echo "Disabling workflow to prevent further runs"
            gh workflow disable "$WORKFLOW_NAME"
          fi
```